### PR TITLE
PYTHON-3797 cache commonly used strings

### PR DESCRIPTION
**Pull request by sleepyStick**
https://jira.mongodb.org/browse/PYTHON-3797

### CODE REVIEW

File: bson/_cbsonmodule.c [hunk 1]
```diff
@@ -70,6 +70,11 @@ struct module_state {
     PyObject* _bid_str;
     PyObject* _replace_str;
     PyObject* _astimezone_str;
+    PyObject* _id_str;
+    PyObject* _$ref_str;
```
Comment (blink1073) [inline bson/_cbsonmodule.c:74]: While these are valid identifiers, they are a bit confusing (and also tripped up the syntax highlighting), can we remove the dollar signs?

**Review by blink1073 — CHANGES_REQUESTED** [review]

File: bson/_cbsonmodule.c (hunk 1 above)
Comment (sleepyStick) [reply bson/_cbsonmodule.c]: yes, fixed!

File: bson/_cbsonmodule.c [hunk 2]
```diff
@@ -70,6 +70,11 @@ struct module_state {
     PyObject* _bid_str;
     PyObject* _replace_str;
     PyObject* _astimezone_str;
+    PyObject* _id_str;
+    PyObject* _ref_str;
```
Comment (blink1073) [inline bson/_cbsonmodule.c:74]: Hmm, now the `__id_str` stands out as odd.  Perhaps use `_dollar_<foo>_str` for each of them?

Comment (sleepyStick) [reply bson/_cbsonmodule.c]: good idea! just changed it

**Review by ShaneHarvey — APPROVED** [review]
Nice work, LGTM!

**Pull request merged**